pkg/operator/resource/resourceapply/apps.go

```diff
@@ -94,6 +94,7 @@ func ApplyDaemonSet(client appsclientv1.DaemonSetsGetter, recorder events.Record
 			toWrite.Spec.Template.Annotations = map[string]string{}
 		}
 		toWrite.Annotations["operator.openshift.io/force"] = forceString
+		toWrite.Annotations["operator.openshift.io/pull-spec"] = toWrite.Spec.Template.Spec.Containers[0].Image
```

Comment:
open to recommendations on the annotation name since it's now reused by different components.